lib/vanagon/platform/windows.rb

```diff
@@ -17,21 +45,42 @@ def generate_package(project)
       def generate_packaging_artifacts(workdir, name, binding)
-        win_dir = File.join(workdir, "windows")
-        FileUtils.mkdir_p(win_dir)
+        # templates that do require a name change
+        erb_file(File.join(VANAGON_ROOT, "templates/windows/project.nuspec.erb"), File.join(workdir, "#{name}.nuspec"), false, { :binding => binding })
 
-        # We'll uncomment this once we have templates to parse here
-        [""].each do |win_file|
-          #erb_file(File.join(VANAGON_ROOT, "templates/win/#{win_file}.erb"), File.join(win_dir, win_file), false, { :binding => binding })
+        # templates that don't require a name change
+        ["chocolateyInstall.ps1", "chocolateyUninstall.ps1"].each do |win_file|
+          erb_file(File.join(VANAGON_ROOT, "templates/windows/#{win_file}.erb"), File.join(workdir, win_file), false, { :binding => binding })
         end
       end
 
+      # The specific bits used to generate a windows nuget package for a given project
+      # Nexus expects packages to be named #{name}-#{version}.nupkg. However, chocolatey
+      # will generate them to be #{name}.#{version}.nupkg. So, we have to rename the
+      # package after we build it. We also take advantage of the rename to sneak the
+      # architecture in the package name. Neither chocolatey nor nexus know how to deal
+      # with architecture, so we are just pretending it's part of the package name.
+      #
+      # @param project [Vanagon::Project] project to build a nuget package of
+      # @return [Array] list of commands required to build a nuget package for the given project from a tarball
+      def generate_nuget_package(project)
+        target_dir = project.repo ? output_dir(project.repo) : output_dir
+        ["mkdir -p output/#{target_dir}",
+        "mkdir -p $(tempdir)/#{project.name}/tools",
+        "cp #{project.name}.nuspec $(tempdir)/#{project.name}/",
+        "cp chocolateyInstall.ps1 chocolateyUninstall.ps1 $(tempdir)/#{project.name}/tools/",
+        "cp file-list $(tempdir)/#{project.name}/tools/file-list.txt",
+        "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/#{project.name}/tools' --strip-components 1 -xf -",
+        "(cd $(tempdir)/#{project.name} ; #{self.drive_root}/ProgramData/chocolatey/bin/choco.exe pack #{project.name}.nuspec)",
+        "cp $(tempdir)/#{project.name}/#{project.name}-#{@architecture}.#{project.version}.#{project.release}.nupkg ./output/#{target_dir}/#{nuget_package_name(project)}"]
+      end
+
       # Method to derive the package name for the project
       #
       # @param project [Vanagon::Project] project to name
       # @return [String] name of the windows package for this project
-      def package_name(project)
-        "#{project.name}-#{project.version}-#{project.release}-#{@architecture}.msi"
+      def nuget_package_name(project)
+        "#{project.name}-#{@architecture}-#{project.version}.#{project.release}.nupkg"
```

Comment:
Check out the comment above `generate_nuget_package`. That has the explanation. I'll add it here too though, just to be thorough